inbm/dispatcher-agent/dispatcher/workload_orchestration_2.py

```diff
@@ -0,0 +1,296 @@
+"""
+    Workload Orchestration methods called by dispatcher before and after
+    any OTA commands having shutdown/reboot.
+
+    Copyright (C) 2017-2024 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+import json
+import logging
+import platform
+import socket
+import time
+from typing import Tuple, Dict, Any, Optional
+from threading import Thread
+from json.decoder import JSONDecodeError
+from os import path
+
+import requests
+from inbm_common_lib.utility import get_canonical_representation_of_path
+from inbm_lib.count_down_latch import CountDownLatch
+from inbm_common_lib.shell_runner import PseudoShellRunner
+
+from .config.config_command import ConfigCommand
+from .constants import *
+from .dispatcher_exception import DispatcherException
+from .dispatcher_broker import DispatcherBroker
+
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+class WorkloadOrchestration:
+    """Workload Orchestration class to switch device status to schedulable or unschedulable
+
+    @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+    """
+
+    def __init__(self,
+                 dispatcher_broker: DispatcherBroker) -> None:
+        self._dispatcher_broker = dispatcher_broker
+
+    def is_workload_service_file_present(self) -> bool:
+        """Checks if workload orchestration files are present.
+
+        @return: return True if present else False; always False if not on Linux
+        """
+        if platform.system() != 'Linux':
+            return False
+
+        file_path = self.get_orchestrator_value(IP)
+        return True if file_path and path.exists(file_path) else False
+
+    def is_workload_service_active(self) -> bool:
+        """Checks if workload orchestration is active or not.
+
+        @return: return True if service active else False;
+        """
+        orchestrator_service = self.get_orchestrator_value(ORCHESTRATOR)
+        command = f"systemctl is-active {orchestrator_service}"
+        (out, err, code) = PseudoShellRunner().run(command)
+        if code == 0 and out.strip() == 'active':
+            self._dispatcher_broker.telemetry(
+                "Workload Orchestration Service Active")
+            return True
+        else:
+            return False
+
+    def set_workload_orchestration_mode(self, online_mode: bool) -> None:
+        """Changes the workload orchestration device status.
+
+        @param online_mode: boolean to indicate device to be online mode
+        """
+        if self.is_workload_service_file_present():
+            if online_mode:
+                worker = Thread(target=self._switch_to_online_mode)
+                worker.setDaemon(True)
+                worker.start()
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

It's recommended to abstract the threading logic in `set_workload_orchestration_mode` into a separate method or class that handles thread creation and management. This encapsulation would improve the modularity of the code and make it easier to extend or modify the threading behavior in the future.